src/functional/ffront/parsing/linear_assign.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2021, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import ast
+import copy
+from typing import Iterator
+
+
+class NodeYielder(ast.NodeTransformer):
+    @classmethod
+    def mutate_ast(cls, node: ast.AST) -> ast.AST:
+        result = list(cls().visit(node))
+        if len(result) != 1:
+            raise ValueError("AST was split or lost during the pass. Use `.visit()` instead.")
+        return result[0]
+
+    def visit(self, node: ast.AST) -> Iterator[ast.AST]:
+        result = super().visit(node)
+        if isinstance(result, ast.AST):
+            yield result
+        else:
+            yield from result
+
+
+class SingleAssignTargetPass(NodeYielder):
+    """
+    Split multi target assignments.
+
+    Requires AST in SSA form (see ``SingleStaticAssignPass``) to yield correct results.
+
+    Example
+    -------
+    Usage:
+
+        import ast
+        from functional.ffront.parsing.parsing import get_ast_from_func
+
+        def foo():
+            a = b = 1
+            return a, b
+
+        print(ast.unparse(
+            SingleStaticAssignPass.mutate_ast(
+                get_ast_from_func(foo)
+            )
+        ))
+
+        # will print out:
+
+        def foo():
+            a = 1
+            b = 1
+
```

Comment:
I would use doctest syntax in the docstring to make sure it never gets out of sync and it is nicely formatted in the docs. (See several examples here: https://github.com/GridTools/gt4py/blob/master/src/eve/utils.py)